source/dofs/dof_handler_policy.cc

```diff
@@ -3263,10 +3278,10 @@ namespace internal
 
 
 
-        template <int dim, int spacedim>
+        template <typename DoFHandlerType>
         void
         communicate_dof_indices_on_marked_cells
-        (const DoFHandler<dim,spacedim> &dof_handler,
+        (const DoFHandlerType &dof_handler,
```

Comment:
I am again in favour of restricting this template to `DoFHandlerType<dim, spcedim>`, i.e.
```
         template <int dim, int spacedim, template <int, int> typename DoFHandlerType>
         void         
         communicate_dof_indices_on_marked_cells
         (const DoFHandlerType<dim, spacedim> &dof_handler,
```